File: helpdesk-pro/frontend/src/app/core/services/canned-response.service.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';
import { map } from 'rxjs/operators';
import { environment } from '../../../environments/environment';
import { ApiResponse } from '../models';

export interface CannedResponse {
  id: string;
  title: string;
  body: string;
  category?: string;
  createdAt: string;
}

export interface CannedResponseRequest {
  title: string;
  body: string;
  category?: string;
}

@Injectable({ providedIn: 'root' })
export class CannedResponseService {
  private base = `${environment.apiUrl}/canned-responses`;

  constructor(private http: HttpClient) {}

  getAll(): Observable<CannedResponse[]> {
    return this.http.get<ApiResponse<CannedResponse[]>>(this.base).pipe(map(r => r.data));
  }

  create(data: CannedResponseRequest): Observable<CannedResponse> {
    return this.http.post<ApiResponse<CannedResponse>>(this.base, data).pipe(map(r => r.data));
  }

  update(id: string, data: CannedResponseRequest): Observable<CannedResponse> {
    return this.http.put<ApiResponse<CannedResponse>>(`${this.base}/${id}`, data).pipe(map(r => r.data));
  }

  delete(id: string): Observable<void> {
    return this.http.delete<void>(`${this.base}/${id}`);
  }
}

File: helpdesk-mobile/lib/core/services/canned_response_service.dart
import '../api/api_client.dart';
import '../api/api_endpoints.dart';
import '../models/canned_response_model.dart';

class CannedResponseService {
  final ApiClient _api;
  CannedResponseService(this._api);

  Future<List<CannedResponseModel>> getAll() async {
    final resp = await _api.get(ApiEndpoints.cannedResponses);
    final data = resp.data['data'];
    final list = data is List ? data : (data['content'] ?? []);
    return List<CannedResponseModel>.from(list.map((e) => CannedResponseModel.fromJson(e)));
  }

  Future<CannedResponseModel> create(Map<String, dynamic> data) async {
    final resp = await _api.post(ApiEndpoints.cannedResponses, data: data);
    return CannedResponseModel.fromJson(resp.data['data']);
  }

  Future<CannedResponseModel> update(String id, Map<String, dynamic> data) async {
    final resp = await _api.put('${ApiEndpoints.cannedResponses}/$id', data: data);
    return CannedResponseModel.fromJson(resp.data['data']);
  }

  Future<void> delete(String id) async {
    await _api.delete('${ApiEndpoints.cannedResponses}/$id');
  }
}
